services/dynamic/dynamic-xml.service.js

```diff
@@ -47,14 +36,48 @@ module.exports = class DynamicXml extends BaseService {
     } catch (e) {
       throw new InvalidParameter({ prettyMessage: e.message })
     }
-    values = values.map((node, i) =>
-      pathIsAttr ? node.value : node.firstChild.data
-    )
+
+    if (!Array.isArray(values)) {
+      throw new InvalidResponse({
+        prettyMessage: 'must be a collection of elements',
```

Comment:
At this stage we know that an xpath expression and an XML content are valid, but we are not able to return a result. What do you think about a `unsupported query` message?

Reply:
I'm not familiar with the `xpath` lib we are using to know what would result in the `select` function returning a non-array, but `unsupported query` sounds better to me anyway 😄